iroh-gossip/src/proto/mod.rs

```diff
@@ -0,0 +1,29 @@
+use std::{fmt, hash::Hash};
+
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+pub mod hyparview;
+pub mod plumtree;
```

Comment:
it would be great to have a highlevel description somewhere how plumtree and hyparview are setup to work together/integrate

Reply:
I added a high level description to the `proto` module docs. Can be extended further for sure, but now there's something there at least.